lib/authorise.js

```diff
@@ -127,13 +130,27 @@ function checkToken (done) {
     self.req.oauth = { bearerToken: token };
     self.req.user = token.user ? token.user : { id: token.userId };
 
-    if(self.scope) {
-	  self.model.checkScope(self.scope, token, function (err) {
-	    if (err) { return done(new error('invalid_scope', err)); }
-	    done();
-	  });
-    } else {
-      done();
+    done();
+  });
+}
+
+/**
+ * Check scope
+ *
+ * @param {Function} done
+ * @this  OAuth
+ */
+
+function checkScope (done) {
+  if (!this.options.scope) return done();
+
+  this.model.authoriseScope(this.req.oauth.bearerToken, this.options.scope, function (err, allowed) {
```

Comment:
At this point we're attempting to access a protected route with an access token that has already been granted some scope. Thus if `model.authoriseScope` returns a falsy allowed, the error message should reflect that we lack sufficient privileges. Taking this further, why not change `allowed` to `invalid`? By doing so, `model.authoriseScope` may pass a string describing *what* makes the scope invalid.